src/game/Strategic/StrategicMap.cc

```diff
@@ -770,17 +772,17 @@ void HandleQuestCodeOnSectorEntry( INT16 sNewSectorX, INT16 sNewSectorY, INT8 bN
 					fred.sSectorX = sNewSectorX;
 					fred.sSectorY = sNewSectorY;
 					fred.bSectorZ = 0;
-					fred.bTown    = gMineLocation[ubThisMine].bAssociatedTown;
+					fred.bTown    = thisMine->associatedTownId;
 
 					// mark miner as placed
 					ubRandomMiner[ 0 ] = 0;
 					ubMinersPlaced++;
 				}
 
 				// assign the remaining (3) miners randomly
-				for ( ubMine = 0; ubMine < MAX_NUMBER_OF_MINES; ubMine++ )
+				for ( ubMine = 0; ubMine < GCM->getMines().size(); ubMine++ )
 				{
-					if ( ubMine == ubThisMine || ubMine == MINE_ALMA || ubMine == MINE_SAN_MONA )
+					if ( ubMine == ubThisMine || ubMine == MINE_ALMA || thisMine->isAbandoned() )
```

Comment:
`thisMine->isAbandoned()` can't be true here, since we're still in the original block. Should be checking via ubMine.

Reply:
Can we do even better though? If someone adds two more mines, should they appear here or not? Either way, wouldn't another field, eg. `hasHeadMiner` solve it in a simpler manner?